core/spec/models/spree/order_spec.rb

```diff
@@ -436,8 +436,74 @@ def compute(computable)
         expect{ order.restart_checkout_flow }.to change{order.state}.from("delivery").to("cart")
       end
     end
+
+    context "line items discontinued" do
+      let(:order) { create :order_with_line_items, line_items_count: 1, state: "delivery" }
+
+      before { order.line_items.first.variant.discontinue! }
+
+      it "updates the state column to cart" do
+        expect{ order.restart_checkout_flow }.to change{order.state}.from("delivery").to("cart")
+      end
+    end
+
+    context "line items have insufficient stock" do
+      let(:order) { create :order_with_line_items, line_items_count: 1, state: "delivery" }
+
+      before do
+        order.line_items.first.variant.stock_items.update_all(backorderable: false)
+      end
+
+      it "updates the state column to cart" do
+        expect{ order.restart_checkout_flow }.to change{order.state}.from("delivery").to("cart")
```

Comment:
Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.